### PR TITLE
docs(src/*): Add TSDoc comments to improve code documentation

### PR DESCRIPTION
**Pull request by fzn0x**
Port of #2677 and better.

### The author should do the following, if applicable

- [ ] Add tests
- [ ] Run tests
- [ ] `bun run format:fix && bun run lint:fix` to format the code

### CODE REVIEW

File: src/request.ts [hunk 1]
```diff
@@ -265,18 +265,50 @@ export class HonoRequest<P extends string = '/', I extends Input['out'] = {}> {
...
 
+  /**
+   * Parses the request body as a `Blob`.
+   * @example
+   * ```ts
+   * app.post('/entry', async (c) => {
+   *   const body = await c.req.blob();
+   * });
+   * ```
+   * @see https://hono.dev/api/request#blob
```
Comment (fzn0x) [inline src/request.ts:276]: TODO: add documentation for blob in hono/website

**Comment by fzn0x** [comment]
PR is ported @yusukebe thanks for your patience!

File: src/request.ts [hunk 2]
```diff
@@ -265,18 +265,50 @@ export class HonoRequest<P extends string = '/', I extends Input['out'] = {}> {
...
 
+  /**
+   * Parses the request body as `FormData`.
+   * @example
+   * ```ts
+   * app.post('/entry', async (c) => {
+   *   const body = await c.req.formData();
+   * });
+   * ```
+   * @see https://hono.dev/api/request#formdata
```
[the remark this answers is not in the document]
Comment (fzn0x) [reply src/request.ts:290]: Resolved

File: src/request.ts (hunk 1 above)
Comment (fzn0x) [reply src/request.ts:276]: Resolved

File: src/request.ts [hunk 3]
```diff
@@ -265,18 +265,52 @@ export class HonoRequest<P extends string = '/', I extends Input['out'] = {}> {
...
     return this.cachedBody('formData')
   }
 
+  /**
+   * Adds validated data to the request.
+   *
+   * @param target - The target of the validation.
+   * @param data - The validated data to add.
+   */
   addValidatedData(target: keyof ValidationTargets, data: {}) {
```
Comment (fzn0x) [inline src/request.ts:302]: I believe we should document this in hono/website too

I will try the feature first and work on it! @yusukebe 

Looks like adding TSDoc/JSDoc helps to discover undocumented features!

File: src/hono-base.ts [hunk 4]
```diff
@@ -83,14 +85,65 @@ export type HonoOptions<E extends Env> = {
 
 class Hono<E extends Env = Env, S extends Schema = {}, BasePath extends string = '/'> {
   // Theses methods are dynamically initialized in the constructor.
+  /**
```
Comment (yusukebe) [inline src/hono-base.ts:88]: It seems these docs are not used.

CleanShot 2024-05-29 at 02 05 40@2x

I don't have a good idea to show the docs for the super complex types in the IDE. But, I think these docs for `get!` or `post!` etc. are not necessary.

Comment (fzn0x) [reply src/hono-base.ts]: Resolved!

File: src/hono-base.ts [hunk 5]
```diff
@@ -3,7 +3,6 @@
  * This module is the base module for the Hono object.
  */
 
-/* eslint-disable @typescript-eslint/no-explicit-any */
```
[the remark this answers is not in the document]
Comment (fzn0x) [reply src/hono-base.ts]: Resolved!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@fzn0x 

Thanks! Looks good to me. We may find improvements in the future, but we will address them at that time.

**Pull request merged**